src/config.js

```diff
@@ -18,18 +18,25 @@ const getRef = () => {
     }
     default: {
       core.warning(`Received unexpected github event ${github.context.eventName}`);
-      return github.context.payload?.ref;
+      return github.context.payload?.ref || github.context.ref;
     }
   }
 };
 
 const thisBranchName = () => {
 
-  const ref = getRef();
+  try {
+    const ref = getRef();
+
+    const refParts = ref.split('/');
 
-  const refParts = ref.split('/');
+    return refParts[refParts.length-1];
+  }
+  catch (error) {
+    core.error(`Unable to get current branch name : ${error.message}`);
```

Comment:
Might want to include the ref in the error message - it's the thing that u try to split and blows up...